test/models/work_test.rb

```diff
@@ -0,0 +1,225 @@
+      expect(repeat_work.errors.messages).must_include :title
+      expect(repeat_work.errors.messages[:title]).must_equal ["has already been taken"]
+      # new work
+      expect(works(:new_cat_work).valid?).must_equal true
+      expect(works(:new_cat_work).errors.messages).must_be_empty
+    end
+
+    it "must have a category" do
+      # Arrange
+      works(:book).category = nil
+
+      # Assert
+      expect(works(:book).valid?).must_equal false
+      expect(works(:book).errors.messages).must_include :category
+      # note: even though they get the same message,
+      # nil vs a wrong category still have to be tested separately
+      expect(works(:book).errors.messages[:category]).must_equal ["must be book, movie, or album"]
+    end
+
+    it "accepts valid categories regardless of case, converting to downcase" do
+      caps_work = Work.create(category: "MOVIE", title: "Movie")
+      # Assert
+      expect(caps_work.valid?).must_equal true
+      expect(caps_work.errors.messages).must_be_empty
+      expect(caps_work.category).must_equal 'movie'
+    end
+
+    it "rejects invalid category types as a safeguard" do
+      # Arrange
+      works(:book).category = 'FAKER'
+
+      # Assert
+      expect(works(:book).valid?).must_equal false
+      expect(works(:book).errors.messages).must_include :category
+      # note: even though they get the same message,
+      # nil vs a wrong category still have to be tested separately
+      expect(works(:book).errors.messages[:category]).must_equal ["must be book, movie, or album"]
+    end
+
+    # we know the behavior of integer typecasting
+    # but we also want to document the behavior from a user POV
+    it "converts float years to integers, strings to integer" do
+      # try a float
+      works(:book).publication_year = 2.44
+      expect(works(:book).publication_year).must_equal 2
+
+      # try a string
+      works(:book).publication_year = 'NOTAYEAR'
+      expect(works(:book).publication_year).must_equal 0
+
+      # try a numerical string
+      works(:book).publication_year = '1999'
+      expect(works(:book).publication_year).must_equal 1999
+    end
+  end
+
+  describe "custom methods" do
+    before do
+      @user = User.create(username: 'voter')
+      @newer_work = Work.new(category: 'movie', title: 'A')
+    end
+
+    describe "spotlight" do
+      before do
+        Vote.delete_all # for this particular section we need to manipulate votes a certain way
+      end
+      it "returns nil if there are no works" do
+        Work.delete_all
+        assert_nil(Work.spotlight)
+      end
+      it "returns work with most votes" do
+        # @work1.save
+        # @work2.save
+
+        Vote.create(work_id: works(:workBB).id, user_id: @user.id)
+
+        expect(Work.spotlight).must_equal works(:workBB)
+        expect(Work.spotlight.votes.count).must_equal 1
+        expect(Work.spotlight.votes.count).must_equal works(:workBB).votes.count
```

Comment:
Going off of the longer comment from before, since we've tested that `Work`s indeed have votes, this specific line might feel like a redundant test-- it's testing that `votes.count` of two arrays is the same.

If we really want to test that a vote count increases, this might feel more like a controller test-- testing that the action of voting increases the vote count.